### PR TITLE
Refactor config

### LINKED ISSUE #4131: 🔧 [Refactor] Config
## :zap: Refactor ticket

Refactor config to define default values and have a centralized place where that all happens.
Do not use `process.env` anymore.

### Motive

Better overview, default values & documentation for configs

### PR DESCRIPTION
**Pull request by ulfgebhardt**
## 🍰 Pullrequest

Refactor the config handling for front- and backend. 

### Issues

- fixes #4131

### Todo

- [ ] The frontend uses two config packages - nuxt has a new way of handling configs and the nuxt.js.config is used for a third option to define config values. This needs to be unified. Make a Ticket before closing.

### CODE REVIEW

File: backend/src/config/index.js [hunk 1]
```diff
@@ -2,111 +2,106 @@ import dotenv from 'dotenv'
...
 
+// Load env file
 if (require.resolve) {
-  // are we in a nodejs environment?
   try {
     dotenv.config({ path: require.resolve('../../.env') })
   } catch (error) {
-    if (error.code !== 'MODULE_NOT_FOUND') throw error
-    console.log('WARN: No `.env` file found in /backend') // eslint-disable-line no-console
+    if (error.code === 'MODULE_NOT_FOUND') {
```
Comment (Mogge) [inline backend/src/config/index.js:10]: Finally someone caught this error. Please give a hint how to fix it: 
`cp backend/.env.template .env`

**Review by Mogge — DISMISSED** [review]
First of all, I get the following warning when starting webapp:
```
 ERROR  [HPM] Missing "target" option. Example: {target: "http://www.example.org"}                                                                                                                                                                                   14:11:47

  at Object.createConfig (node_modules/http-proxy-middleware/lib/config-factory.js:43:11)
  at new HttpProxyMiddleware (node_modules/http-proxy-middleware/lib/index.js:17:30)
  at module.exports (node_modules/http-proxy-middleware/index.js:4:10)
  at node_modules/@nuxtjs/proxy/lib/module.js:63:30
  at Array.forEach (<anonymous>)
  at ModuleContainer.proxyModule (node_modules/@nuxtjs/proxy/lib/module.js:61:9)
  at ModuleContainer.addModule (node_modules/@nuxt/core/dist/core.js:215:34)
  at ModuleContainer.requireModule (node_modules/@nuxt/core/dist/core.js:133:17)
  at ModuleContainer.axiosModule (node_modules/@nuxtjs/axios/lib/module.js:106:10)
  at ModuleContainer.addModule (node_modules/@nuxt/core/dist/core.js:215:34)
```
This message is new.
I can reach localhost:3000 but have the following new warnings inn the console:
```
[Vue warn]: Error in render: "TypeError: this.notifications is undefined"
TypeError: this.notifications is undefined
```
## I can not login!
```
Error Network error: JSON.parse: unexpected character at line 1 column 1 of the JSON data
```
As far as I can see, the backend is running fine (port 4000 and tests) but the webapp is not running (although the tests pass).

**Comment by ulfgebhardt** [comment]
Good catch! It's fixed

**Review by Mogge — APPROVED** [review]
Looks good. Everything is working as expected

**Pull request merged**